Tests/SonarScanner.MSBuild.PreProcessor.Test/WebClientDownloaderTest.cs

```diff
@@ -144,6 +150,112 @@ public void MultipleDisposeCallsNotFailing()
+
+            var sut = new WebClientDownloader("username", "password", logger, httpClient: httpClient);
+            var stream = await sut.DownloadStream(new Uri("https://test.com"), true);
+
+            stream.Should().BeNull();
+
+            logger.AssertDebugLogged("Downloading from https://test.com/...");
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public async Task DownloadStream_Fail_NoLog()
+        {
+            const string testContent = "test content";
+            var logger = new TestLogger();
+            var httpClient = MockHttpClient(new HttpResponseMessage { StatusCode = HttpStatusCode.Forbidden, Content = new StringContent(testContent) });
+
+            var sut = new WebClientDownloader("username", "password", logger, httpClient: httpClient);
+            var stream = await sut.DownloadStream(new Uri("https://test.com"));
+
+            stream.Should().BeNull();
+
+            logger.AssertDebugLogged("Downloading from https://test.com/...");
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public async Task Download_Success()
+        {
+            const string testContent = "test content";
+            var logger = new TestLogger();
+            var httpClient = MockHttpClient(new HttpResponseMessage { StatusCode = HttpStatusCode.OK, Content = new StringContent(testContent) });
+
+            var sut = new WebClientDownloader("username", "password", logger, httpClient: httpClient);
+            var text = await sut.Download(new Uri("https://test.com"));
+
+            text.Should().Be(testContent);
+
+            logger.AssertDebugLogged("Downloading from https://test.com/...");
+        }
+
+        [TestMethod]
+        public async Task Download_Fail_NoLog()
+        {
+            const string testContent = "test content";
+            var logger = new TestLogger();
+            var httpClient = MockHttpClient(new HttpResponseMessage { StatusCode = HttpStatusCode.Forbidden, Content = new StringContent(testContent) });
+
+            var sut = new WebClientDownloader("username", "password", logger, httpClient: httpClient);
+            var text = await sut.Download(new Uri("https://test.com"));
+
+            text.Should().BeNull();
+
+            logger.AssertDebugLogged("Downloading from https://test.com/...");
+            logger.AssertNoWarningsLogged();
+        }
+
+        private static HttpClient MockHttpClient(HttpResponseMessage responseMessage)
+        {
+            var httpMessageHandlerMock = new Mock<HttpMessageHandler>(MockBehavior.Strict);
+            httpMessageHandlerMock
+                .Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(responseMessage)
+                .Verifiable();
```

Comment:
What's this for?
```suggestion
```

Reply:
Not needed here. It would have been verified by the Moq if we would have called `mock.VerifyAll()` or `mock.Verify`